src/figuresClasses.ts

```diff
@@ -1,11 +1,74 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = 'triangle',
+    public trianglePerimeter: number = a + b + c,
```

Comment:
[CODE STYLE] - The calculation of the `trianglePerimeter` should not be done as a default parameter value. It's better to perform this calculation within the constructor body, after the validation checks, to ensure that the sides can indeed form a triangle.